sound/soc/sof/intel/hda.c

```diff
@@ -179,6 +179,40 @@ void hda_dsp_dump(struct snd_sof_dev *sdev, u32 flags)
 	}
 }
 
+void hda_ipc_regs_dump(struct snd_sof_dev *sdev)
+{
+	u32 hipci;
+	u32 hipcie;
+	u32 hipct;
+	u32 hipcte;
+	u32 hipcctl;
+
+	/* read IPC status */
+	hipcie = snd_sof_dsp_read(sdev, HDA_DSP_BAR,
+				  HDA_DSP_REG_HIPCIE);
+	hipct = snd_sof_dsp_read(sdev, HDA_DSP_BAR, HDA_DSP_REG_HIPCT);
+	hipcctl = snd_sof_dsp_read(sdev, HDA_DSP_BAR, HDA_DSP_REG_HIPCCTL);
+
+	hipci = snd_sof_dsp_read(sdev, HDA_DSP_BAR,
+				 HDA_DSP_REG_HIPCI);
+
+	hipcte = snd_sof_dsp_read(sdev, HDA_DSP_BAR,
+				  HDA_DSP_REG_HIPCTE);
+
+	/* dump the IPC regs */
+	/* TODO: parse the raw msg */
+	dev_err(sdev->dev, "error: host state: %s\n", "");
+	dev_err(sdev->dev, "error: host mask: %s\n", "");
+	dev_err(sdev->dev, "error: host raw: status = 0x%8.8x mask = 0x%8.8x\n",
+		hipcie, hipcctl);
```

Comment:
should we use rate_limited versions?

Reply:
what is a rate_limited version here?

Reply:
dev_err_ratelimited()

Reply:
I did not think we need here. Since IPC timed out will be a blocker for the FW, the FW will be dead if an timed out happen.  I think there will not be a flooding.